harness/determined/common/experimental/experiment.py

```diff
@@ -16,9 +21,63 @@ class is obtained via :class:`~determined.experimental.Determined`, the
             master URL is automatically passed into this constructor.
     """
 
-    def __init__(self, experiment_id: int, master: str):
+    def __init__(
+        self,
+        experiment_id: int,
+        master: str,
+        api_ref: ExperimentsApi,
+        config: Optional[Dict] = None,
+    ):
         self.id = experiment_id
         self._master = master
+        self._experiments = api_ref
+        self.config = config
+
+    def activate(self) -> None:
+        self._experiments.determined_activate_experiment(id=self.id)
+
+    def archive(self) -> None:
+        self._experiments.determined_archive_experiment(id=self.id)
+
+    def cancel(self) -> None:
+        self._experiments.determined_cancel_experiment(id=self.id)
+
+    def delete(self) -> None:
+        self._experiments.determined_delete_experiment(id=self.id)
+
+    def kill(self) -> None:
+        self._experiments.determined_kill_experiment(id=self.id)
+
+    def pause(self) -> None:
+        self._experiments.determined_pause_experiment(id=self.id)
+
+    def unarchive(self) -> None:
+        self._experiments.determined_unarchive_experiment(id=self.id)
+
+    def wait_till_complete(self, sleep_interval: int = 5) -> None:
+        """
+        Wait for experiment to reach complete or a terminal state.
+
+        Arguments:
+            sleep_interval (int, optional): An interval time in seconds before checking
+            next experiement state.
+        """
+        while True:
+            exp_resp = self._experiments.determined_get_experiment(experiment_id=self.id)
+            if exp_resp.experiment.state in (
+                Determinedexperimentv1State.COMPLETED,
+                Determinedexperimentv1State.CANCELED,
+                Determinedexperimentv1State.DELETED,
+                Determinedexperimentv1State.ERROR,
+            ):
+                print("Experiment {} is in a terminal state".format(self.id))
+                break
+            elif exp_resp.experiment.state == Determinedexperimentv1State.PAUSED:
+                raise ValueError("Experiment {} is in paused state".format(self.id))
+            else:
+                # ACTIVE, STOPPING_COMPLETED, etc.
+                print("Waiting for Experiment {} to complete".format(self.id))
```

Comment:
let's just leave this out.  Or maybe print it if a minute has passed or something.